### PR TITLE
♻️ chore: remove upgradeTo from UUPSUpgradeable

### PR DESCRIPTION
**Pull request by howydev**
## Description

1. Remove `upgradeTo` from UUPSUpgradeable
2. Update `upgradeToAndCall` to use conditional logic, switching between doing the extra `call` based on `data.length`

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge fmt`?
- [x] Ran `forge snapshot`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

File: src/utils/UUPSUpgradeable.sol [hunk 1]
```diff
@@ -89,29 +94,18 @@ abstract contract UUPSUpgradeable {
...
-        assembly {
-            // Forwards the `data` to `newImplementation` via delegatecall.
-            let m := mload(0x40)
-            calldatacopy(m, data.offset, data.length)
-            if iszero(delegatecall(gas(), newImplementation, m, data.length, codesize(), 0x00)) {
-                // Bubble up the revert if the call reverts.
-                returndatacopy(m, 0x00, returndatasize())
-                revert(m, returndatasize())
+            // Calls `newImplementation` if `data` is non-empty.
+            if iszero(iszero(data.length)) {
```
Comment (Vectorized) [inline src/utils/UUPSUpgradeable.sol:99]: Just `if data.length { .. }` will do.

File: src/utils/UUPSUpgradeable.sol [hunk 2]
```diff
@@ -89,29 +94,18 @@ abstract contract UUPSUpgradeable {
...
-        /// @solidity memory-safe-assembly
-        assembly {
-            // Forwards the `data` to `newImplementation` via delegatecall.
-            let m := mload(0x40)
-            calldatacopy(m, data.offset, data.length)
-            if iszero(delegatecall(gas(), newImplementation, m, data.length, codesize(), 0x00)) {
-                // Bubble up the revert if the call reverts.
-                returndatacopy(m, 0x00, returndatasize())
-                revert(m, returndatasize())
+            // Calls `newImplementation` if `data` is non-empty.
```
Comment (Vectorized) [inline src/utils/UUPSUpgradeable.sol:98]: Comment nit: 

```
Perform a delegatecall to `newImplementation` if `data` is non-empty.  
```

**Review by Vectorized — APPROVED** [review]
LGTM. I'll merge shortly.

**Comment by Vectorized** [comment]
Don't worry about the `__codesize`. It includes the test code, which increases due to the need to call a function with more arguments.

**Comment by howydev** [comment]
changes made, will let you do the merge. thank you senpai :pray:

**Pull request merged**